dataactbroker/handlers/fileHandler.py

```diff
@@ -779,6 +780,75 @@ def check_generation(submission, file_type):
 
         return JsonResponse.create(StatusCode.OK, response_dict)
 
+    @staticmethod
+    def submit_detached_file(submission):
+        """ Submits the FABS upload file associated with the submission ID """
+        # Check to make sure it's a d2 submission
+        if not submission.d2_submission:
+            raise ResponseException("This is not a FABS submission", StatusCode.CLIENT_ERROR)
```

Comment:
We may want to reword these error messages. using "This" is very specific. We may want to say something like "Submission is not a FABS submission". Should probably consult Kaitlin about this depending on how the message is displayed on the front-end as well